multiraft/multiraft.go

```diff
@@ -649,6 +649,11 @@ func (s *state) createGroup(groupID uint64) error {
 		}
 	}
 
+	// Automatically campaign and elect a leader for this group if there's
+	// exactly one known node for this group.
+	if len(cs.Nodes) == 1 && s.MultiRaft.nodeID == NodeID(cs.Nodes[0]) {
```

Comment:
Good point, I've taken care of populating `cs.Nodes` only for initialized or found ranges - see next commit.

As for conf changes, I just looked at the code and (this is off topic) am a little confused - the conf change is applied immediately when it's appended, but there's no place in which the previous state is stored. It's my understanding that a conf change will have to be undone in some cases (for example if it does not commit and someone without it becomes leader), but I don't see that reflected in multinode/raft.